### PR TITLE
diskonaut: init at 0.3.0

### PR DESCRIPTION
**Pull request by evanjs**
###### Motivation for this change
[Diskonaut](https://github.com/imsnif/diskonaut) — A disk usage explorer / terminal disk space navigator 
Featured as crate of the week in **This Week In Rust** 344 ([23 Jun 2020](https://this-week-in-rust.org/blog/2020/06/23/this-week-in-rust-344/))

###### Tests currently failing
```
builder for '/nix/store/7i727mk7dif5jmvh06xrj77162yl8s3k-diskonaut-1.15.1.drv' failed with exit code 101; last 10 log lines:
  thread 'tests::cases::ui::small_files_with_x_as_zero' panicked at 'snapshot assertion for 'small_files_with_x_as_zero' failed in line 2108', /build/diskonaut-1.15.1-vendor.tar.gz/insta/src/runtime.rs:995:9
  
  
  failures:
      tests::cases::ui::small_files_with_x_as_zero
      tests::cases::ui::small_files_with_y_as_zero
  
  test result: FAILED. 35 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out
  
  error: test failed, to rerun pass '--bin diskonaut'
[0 built (1 failed)]
error: build of '/nix/store/7i727mk7dif5jmvh06xrj77162yl8s3k-diskonaut-1.15.1.drv' failed
```

### CODE REVIEW

**Review by iblech — APPROVED** [review]
Very cool, thanks for packaging diskonaut! I justed tested it here, works beautifully. Wouldn't know how to improve the build file, looks perfect. Thank you!

**Comment by iblech** [comment]
Also I should say that no tests are failing on my computer:

         Running target/x86_64-unknown-linux-gnu/release/deps/diskonaut-40e8cbd4e97b6b1c
    
    running 37 tests
    test tests::cases::ui::cant_delete_file_with_term_too_small ... ok
    test tests::cases::ui::clear_selection_when_moving_off_screen_edges ... ok
    test tests::cases::ui::delete_file ... ok
    test tests::cases::ui::delete_file_press_n ... ok
    test tests::cases::ui::delete_folder ... ok
    test tests::cases::ui::delete_folder_small_window ... ok
    test tests::cases::ui::cannot_move_into_small_files ... ok
    test tests::cases::ui::delete_folder_with_multiple_children ... ok
    test tests::cases::ui::empty_folder ... ok
    test tests::cases::ui::eleven_files ... ok
    test tests::cases::ui::enter_folder ... ok
    test tests::cases::ui::enter_folder_medium_width ... ok
    test tests::cases::ui::enter_folder_small_width ... ok
    test tests::cases::ui::files_with_size_zero ... ok
    test tests::cases::ui::medium_width ... ok
    test tests::cases::ui::esc_to_go_up ... ok
    test tests::cases::ui::move_down_and_enter_folder ... ok
    test tests::cases::ui::move_left_and_enter_folder ... ok
    test tests::cases::ui::minimum_tile_sides ... ok
    test tests::cases::ui::move_right_and_enter_folder ... ok
    test tests::cases::ui::move_up_and_enter_folder ... ok
    test tests::cases::ui::noop_when_entering_file ... ok
    test tests::cases::ui::permission_denied_when_deleting ... ok
    test tests::cases::ui::pressing_delete_with_no_selected_tile ... ok
    test tests::cases::ui::noop_when_pressing_esc_at_base_folder ... ok
    test tests::cases::ui::small_files_with_x_as_zero ... ok
    test tests::cases::ui::small_files ... ok
    test tests::cases::ui::small_width ... ok
    test tests::cases::ui::small_width_long_folder_name ... ok
    test tests::cases::ui::too_small_height ... ok
    test tests::cases::ui::too_small_width_five ... ok
    test tests::cases::ui::too_small_width_four ... ok
    test tests::cases::ui::too_small_width_one ... ok
    test tests::cases::ui::too_small_width_three ... ok
    test tests::cases::ui::too_small_width_two ... ok
    test tests::cases::ui::two_large_files_one_small_file ... ok
    test tests::cases::ui::small_files_with_y_as_zero ... ok
    
    test result: ok. 37 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

File: pkgs/tools/misc/diskonaut/default.nix
Comment (danieldk) [inline pkgs/tools/misc/diskonaut/default.nix:5]:
```suggestion
  version = "0.3.0";
```

This version looks wrong? The latest upstream is 0.3.0:

https://github.com/imsnif/diskonaut/releases

**Comment by evanjs** [comment]
> Also I should say that no tests are failing on my computer:
> 
> 
> 
>          Running target/x86_64-unknown-linux-gnu/release/deps/diskonaut-40e8cbd4e97b6b1c
> 
>     
> 
>     running 37 tests
> 
>     test tests::cases::ui::cant_delete_file_with_term_too_small ... ok
> 
>     test tests::cases::ui::clear_selection_when_moving_off_screen_edges ... ok
> 
>     test tests::cases::ui::delete_file ... ok
> 
>     test tests::cases::ui::delete_file_press_n ... ok
> 
>     test tests::cases::ui::delete_folder ... ok
> 
>     test tests::cases::ui::delete_folder_small_window ... ok
> 
>     test tests::cases::ui::cannot_move_into_small_files ... ok
> 
>     test tests::cases::ui::delete_folder_with_multiple_children ... ok
> 
>     test tests::cases::ui::empty_folder ... ok
> 
>     test tests::cases::ui::eleven_files ... ok
> 
>     test tests::cases::ui::enter_folder ... ok
> 
>     test tests::cases::ui::enter_folder_medium_width ... ok
> 
>     test tests::cases::ui::enter_folder_small_width ... ok
> 
>     test tests::cases::ui::files_with_size_zero ... ok
> 
>     test tests::cases::ui::medium_width ... ok
> 
>     test tests::cases::ui::esc_to_go_up ... ok
> 
>     test tests::cases::ui::move_down_and_enter_folder ... ok
> 
>     test tests::cases::ui::move_left_and_enter_folder ... ok
> 
>     test tests::cases::ui::minimum_tile_sides ... ok
> 
>     test tests::cases::ui::move_right_and_enter_folder ... ok
> 
>     test tests::cases::ui::move_up_and_enter_folder ... ok
> 
>     test tests::cases::ui::noop_when_entering_file ... ok
> 
>     test tests::cases::ui::permission_denied_when_deleting ... ok
> 
>     test tests::cases::ui::pressing_delete_with_no_selected_tile ... ok
> 
>     test tests::cases::ui::noop_when_pressing_esc_at_base_folder ... ok
> 
>     test tests::cases::ui::small_files_with_x_as_zero ... ok
> 
>     test tests::cases::ui::small_files ... ok
> 
>     test tests::cases::ui::small_width ... ok
> 
>     test tests::cases::ui::small_width_long_folder_name ... ok
> 
>     test tests::cases::ui::too_small_height ... ok
> 
>     test tests::cases::ui::too_small_width_five ... ok
> 
>     test tests::cases::ui::too_small_width_four ... ok
> 
>     test tests::cases::ui::too_small_width_one ... ok
> 
>     test tests::cases::ui::too_small_width_three ... ok
> 
>     test tests::cases::ui::too_small_width_two ... ok
> 
>     test tests::cases::ui::two_large_files_one_small_file ... ok
> 
>     test tests::cases::ui::small_files_with_y_as_zero ... ok
> 
>     
> 
>     test result: ok. 37 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
> 
> 

Ah. Interesting 🤔 I guess I’ll try `nixpkgs-review` and see if that works.

File: pkgs/tools/misc/diskonaut/default.nix
Comment (danieldk) [inline pkgs/tools/misc/diskonaut/default.nix:11]:
```suggestion
    rev = version;
    sha256 = "0vnmch2cac0j9b44vlcpqnayqhfdfdwvfa01bn7lwcyrcln5cd0z";
```

File: pkgs/tools/misc/diskonaut/default.nix
Comment (danieldk) [inline pkgs/tools/misc/diskonaut/default.nix:16]:
```suggestion
  cargoSha256 = "03hqdg6pnfxnhwk0xwhwmbrk4dicjpjllbbai56a3391xac5wmi6";
```

Remove empty `buildInputs`, this Cargo SHA-256 corresponds to version 0.3.0.

**Review by danieldk — CHANGES_REQUESTED** [review]
I added some suggested changes. Tests fail here as well, but there was an upstream issue for that:

https://github.com/imsnif/diskonaut/issues/42

Until the next release, you can [fetchpatch](https://nixos.org/nixpkgs/manual/#sec-patches) this patch:

https://github.com/imsnif/diskonaut/commit/e94f9b942d8928b7993351a4109cfcef03055d2f.diff

It would also be good to add a comment above the patch with a reference to [the PR](https://github.com/imsnif/diskonaut/pull/43) saying that it can be removed after the next version bump.

**Comment by evanjs** [comment]
> I added some suggested changes. Tests fail here as well, but there was an upstream issue for that:
> 
> 
> 
> https://github.com/imsnif/diskonaut/issues/42
> 
> 
> 
> Until the next release, you can [fetchpatch](https://nixos.org/nixpkgs/manual/#sec-patches) this patch:
> 
> 
> 
> https://github.com/imsnif/diskonaut/commit/e94f9b942d8928b7993351a4109cfcef03055d2f.diff
> 
> 
> 
> It would also be good to add a comment above the patch with a reference to [the PR](https://github.com/imsnif/diskonaut/pull/43) saying that it can be removed after the next version bump.

Sounds good, didn’t see check the issues first. 

And also forgot/missed that version change (I swear I changed it already). Whoops. 

Thanks!

File: pkgs/tools/misc/diskonaut/default.nix
Comment (evanjs) [reply pkgs/tools/misc/diskonaut/default.nix]: Is there any significant difference/reason I should use version over `${version}`?
Am I just hallucinating by remembering seeing `${version}` used before?

**Comment by evanjs** [comment]
Hrm, tests still seem to fail.
@danieldk any thoughts on what I might have missed?

Also: do you think that using the `<PR>.patch` url might be enough to imply "remove this after it has been released"?

File: pkgs/tools/misc/diskonaut/default.nix
Comment (danieldk) [reply pkgs/tools/misc/diskonaut/default.nix]: First of all, since `version` is already a string, they are equivalent, so `version == "${version}"`. However, the former is simpler and the latter is more work for the interpreter (since it has to process the antiquotation).

In the case of versions using a string with antiquotation can be useful when the tag contains the version number as an affix. For example, many projects prefix git version tags with *v*, so in that case you could use:

```
rev = "v${version}";
```

**Comment by danieldk** [comment]
> Hrm, tests still seem to fail.
> @danieldk any thoughts on what I might have missed?

They still fail, because diskonaut reports incorrect file sizes on some filesystems. I have filed an upstream bug:

https://github.com/imsnif/diskonaut/issues/50

File: pkgs/tools/misc/diskonaut/default.nix
Comment (evanjs) [reply pkgs/tools/misc/diskonaut/default.nix]: > First of all, since `version` is already a string, they are equivalent, so `version == "${version}"`. However, the former is simpler and the latter is more work for the interpreter (since it has to process the antiquotation).
> 
> 
> 
> In the case of versions using a string with antiquotation can be useful when the tag contains the version number as an affix. For example, many projects prefix git version tags with *v*, so in that case you could use:
> 
> 
> 
> ```
> 
> rev = "v${version}";
> 
> ```

Ahhh, `v${version}` is the exact case I was thinking of.
The dots in my head just sort of fail to connect at times 🤪
Thanks!

**Comment by evanjs** [comment]
> > Hrm, tests still seem to fail.
> 
> > @danieldk any thoughts on what I might have missed?
> 
> 
> 
> They still fail, because diskonaut reports incorrect file sizes on some filesystems. I have filed an upstream bug:
> 
> 
> 
> https://github.com/imsnif/diskonaut/issues/50

Thanks!

I’ll just skip those specific tests for now, then?

**Comment by evanjs** [comment]
Disabled the failing tests and fixed the hash for the `fetchpatch`.
Let me know if there's anything else!

File: pkgs/tools/misc/diskonaut/default.nix
Comment (danieldk) [inline pkgs/tools/misc/diskonaut/default.nix:28]: Just disable tests altogether until the next release. aarch64 also still fails tests.

```suggestion
  doCheck = false;
```

File: pkgs/tools/misc/diskonaut/default.nix
Comment (danieldk) [inline pkgs/tools/misc/diskonaut/default.nix:20]:
```suggestion
```

With tests disabled this can go.

**Review by danieldk — CHANGES_REQUESTED** [review]
I think at this point it makes more sense to disable tests altogether. However, since incorrect file sizes are reported on the majority of systems that we tested on (yours, mine, the ofborg builders), it probably makes most sense to wait until the next release is out, which hopefully fixes this error and then update this PR for that release.

**Comment by danieldk** [comment]
> it probably makes most sense to wait until the next release is out

Since this is the intended reported usage, let's just disable tests. Maybe you could do that (as in the suggested changes), make this PR a non-draft, and then this PR should be good to go.

**Review by danieldk — APPROVED** [review]
LGTM. I have moved `cargoSha256` up to make the derivation more canonical.

Result of `nixpkgs-review pr 91460` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- diskonaut
</details>

Thanks for making all the requested changes! I'll merge this once ofborg is done.

**Pull request merged**
